### PR TITLE
fix: binops with flags were being converted silently by dropping flags

### PR DESCRIPTION
**Pull request by goens**

### CODE REVIEW

File: language.py [hunk 1]
```diff
@@ -1498,6 +1498,8 @@ def to_lean_binop(bop, state):
   bitwidth = unify_bitwidths([lv1.expr.bw(), lv2.expr.bw()])
   propagate_bitwidth(lv1.expr, bitwidth)
   propagate_bitwidth(lv2.expr, bitwidth)
+  if bop.flags != []:
```
Comment (bollu) [inline language.py:1501]:
```suggestion
  if bop.flags:
```

File: language.py [hunk 2]
```diff
@@ -1498,6 +1498,8 @@ def to_lean_binop(bop, state):
...
+    raise RuntimeError("binop with flags not supported")
```
Comment (bollu) [inline language.py]:
```suggestion
    raise RuntimeError("binops with flags '%s' not supported" % (bop.flags))
```

This should help us incrementally add more flags, so the error message that is logged in `Alive.csv` can tell us what flags are missing.

**Review by bollu — APPROVED** [review]
LGTM

File: language.py (hunk 1 above)
Comment (goens) [reply language.py:1501]: that sounds like a bad idea, no? it makes the code less readable/more opaque

**Pull request merged**